gridcomps/History/MAPL_HistoryGridComp.F90

```diff
@@ -4731,26 +4731,34 @@ subroutine MAPL_SetExpression(nfield,fields,tmpfields,rewrite,nPExtraFields, &
 ! rather than the actual output field variables (i.e., fields(1,:)).
 ! Also do check that there are no illegal operations
 !-------------------------------------------------------------------
+  allocate ( exptmp (1), stat=status )
+  _VERIFY(STATUS)
+  exptmp(1) = ExpState
   ! check which fields are actual exports or expressions
   nPExtraFields = 0
   iRealFields = 0
   allocate(isBundle(nfield))
   do m=1,nfield
-    if (scan(trim(fields(1,m)),'()^*/+-.')/=0) then
-       rewrite(m)= .TRUE.
-       tmpfields(m)= trim(fields(1,m))
-    else
-       if (index(fields(1,m),'%') == 0) then
+
+    call MAPL_ExportStateGet(exptmp,fields(2,m),state,rc=status)
+    _VERIFY(STATUS)
+    if (index(fields(1,m),'%') == 0) then
+       call ESMF_StateGet(state,fields(1,m),field,rc=status)
+       if (status==_SUCCESS) then
           iRealFields = iRealFields + 1
           rewrite(m)= .FALSE.
           isBundle(m) = .FALSE.
           tmpfields(m)= trim(fields(1,m))
-       else
-          isBundle(m)=.true.
-          rewrite(m)= .FALSE.
+       else 
+          rewrite(m)= .TRUE.
```

Comment:
This conditional branch no longer provides a value for isBundle(m)

Reply:
I fixed that.